internal/clients/team3/judge.go

```diff
@@ -11,4 +14,139 @@ type judge struct {
+				tempReturn.Rules = append(tempReturn.Rules, rules.RulesInPlay[rule])
+				tempReturn.Evaluations = append(tempReturn.Evaluations, true)
+			}
+			outMap[clientID] = tempReturn
+		} else {
+			// All other islands will be evaluated fairly using base implementation.
+			tempReturn := outMap[clientID]
+			for _, rule := range rulesAffected {
+				evaluation, err := rules.BasicBooleanRuleEvaluator(rule)
+				if err != nil {
+					return outMap, false
+				}
+				tempReturn.Rules = append(tempReturn.Rules, rules.RulesInPlay[rule])
+				tempReturn.Evaluations = append(tempReturn.Evaluations, evaluation)
+			}
+			outMap[clientID] = tempReturn
+		}
+	}
+	return outMap, true
+}
+
+// CallPresidentElection sets the election settings for the next president election
+func (j *judge) CallPresidentElection(monitoring shared.MonitorResult, turnsInPower int, allIslands []shared.ClientID) shared.ElectionSettings {
+	var electionsettings = shared.ElectionSettings{
+		VotingMethod:  shared.Plurality,
+		IslandsToVote: allIslands,
+		HoldElection:  false,
+	}
+
+	// Base implementation calls an election if monitoring was performed and the result was negative
+	// or if the number of turnsInPower exceeds 3
+	if monitoring.Performed && !monitoring.Result {
+		electionsettings.HoldElection = true
+	}
+	// TODO: think if we want to change strategy here
+	if turnsInPower >= 2 {
+		electionsettings.HoldElection = true
+	}
+	return electionsettings
+}
+
+// DecideNextPresident declares who the next president will be
+func (j *judge) DecideNextPresident(winner shared.ClientID) shared.ClientID {
+	// If the election winner's trust score is high, we will declare them as the next President.
+	// If not, we will replace it with the island who's trust score is the highest.
+	if j.c.trustScore[winner] < 70 {
+		// we can change this to be mailicious everytime
+		for island := range j.c.trustScore {
+			if j.c.trustScore[island] > j.c.trustScore[winner] {
+				winner = island
+			}
+		}
+	}
+	return winner
+}
+
+// GetRuleViolationSeverity returns a custom map of named rules and
+// how severe the sanction should be for transgressing them
+// If a rule is not named here, the default sanction value added is 1
+func (j *judge) GetRuleViolationSeverity() map[string]roles.IIGOSanctionScore {
+	return map[string]roles.IIGOSanctionScore{}
+}
+
+// GetSanctionThresholds returns a custom map of sanction score thresholds for different sanction tiers
+// For any unfilled sanction tiers will be filled with default values (given in judiciary.go)
+func (j *judge) GetSanctionThresholds() map[roles.IIGOSanctionTier]roles.IIGOSanctionScore {
+	return j.BaseJudge.GetSanctionThresholds()
+}
+
+// GetPardonedIslands decides which islands to pardon i.e. no longer impose sanctions on
+// COMPULSORY: decide which islands, if any, to forgive
+func (j *judge) GetPardonedIslands(currentSanctions map[int][]roles.Sanction) map[int][]bool {
+	return j.BaseJudge.GetPardonedIslands(currentSanctions)
```

Comment:
the base judge implementation doesn't pardon any islands ... might be nice to have an internal forgiveness system and pardon some

Reply:
Sure! We are working with @Neelesh99 on this, he suggested similar interesting ideas! :)